src/Screens/Flows/App/AssetDetailScreenSheet/Components/AssetStats.tsx

```diff
@@ -0,0 +1,207 @@
+import * as Haptics from "expo-haptics"
+import { useBrowserTab } from "~Hooks/useBrowserTab"
+import { Routes } from "~Navigation"
+import { SocialLinksButtons } from "./SocialLinksButtons"
+
+const DESCRIPTION_WORD_THRESHOLD = 20
+
+type AssetStatsProps = {
+    tokenSymbol: string
+    tokenDescription?: string
+    socialLinks?: {
+        website?: string
+        twitter?: string
+        telegram?: string
+    }
+}
+
+type StatItem = {
+    label: string
+    value: string | null
+    testID: string
+}
+
+export const AssetStats = ({ tokenSymbol, tokenDescription, socialLinks }: AssetStatsProps): JSX.Element => {
+    const { styles, theme } = useThemedStyles(baseStyles)
+    const { LL } = useI18nContext()
+    const [isAccordionOpen, setIsAccordionOpen] = useState(false)
+    const { navigateWithTab } = useBrowserTab(Routes.HOME)
+
+    const handleSocialLinkPress = useCallback(
+        (url: string) => {
+            navigateWithTab({ url, title: url })
+        },
+        [navigateWithTab],
+    )
+
+    const currency = useAppSelector(selectCurrency)
+
+    const { data: tokenInfo } = useTokenInfo({
+        id: getCoinGeckoIdBySymbol[tokenSymbol],
+    })
+
+    const marketInfo: MarketInfo | undefined = useMemo(() => {
+        if (!tokenInfo) return undefined
+        return {
+            marketCap: tokenInfo?.market_data?.market_cap[currency.toLowerCase()],
+            totalSupply: tokenInfo?.market_data?.total_supply,
+            totalVolume: tokenInfo?.market_data?.total_volume[currency.toLowerCase()],
+            circulatingSupply: tokenInfo?.market_data?.circulating_supply,
+            high24h: tokenInfo?.market_data?.high_24h[currency.toLowerCase()],
+            low24h: tokenInfo?.market_data?.low_24h[currency.toLowerCase()],
+        }
+    }, [tokenInfo, currency])
+
+    const { marketCap, totalSupply, totalVolume, circulatingSupply, high24h, low24h } = useFormattedMarketInfo({
+        marketInfo,
+    })
+
+    const stats: StatItem[] = useMemo(
+        () => [
+            { label: LL.COMMON_MARKET_CAP(), value: marketCap, testID: "stat-market-cap" },
+            { label: LL.COMMON_CIRCULATING_SUPPLY(), value: circulatingSupply, testID: "stat-circulating-supply" },
+            { label: LL.COMMON_TOTAL_SUPPLY(), value: totalSupply, testID: "stat-total-supply" },
+            { label: LL.COMMON_24H_VOLUME(), value: totalVolume, testID: "stat-24h-volume" },
+            { label: LL.COMMON_24H_PRICE_RANGE(), value: `${low24h} - ${high24h}`, testID: "stat-24h-range" },
+        ],
+        [LL, marketCap, circulatingSupply, totalSupply, totalVolume, high24h, low24h],
+    )
+
+    const shouldShowAccordion = useMemo(() => {
+        if (!tokenDescription) return false
+        const wordCount = tokenDescription.trim().split(/\s+/).length
+        return wordCount > DESCRIPTION_WORD_THRESHOLD
+    }, [tokenDescription])
```

Comment:
I'd honestly count the lines instead of this.
There should be an event called `onTextLayout` that will give you back the number of lines used to render the value